cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -334,17 +335,23 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 
 			scaleDownStart := time.Now()
 			metrics.UpdateLastTime(metrics.ScaleDown, scaleDownStart)
-			result, typedErr := scaleDown.TryToScaleDown(allNodes, allScheduled, pdbs, currentTime)
+			scaleDownStatus, typedErr := scaleDown.TryToScaleDown(allNodes, allScheduled, pdbs, currentTime)
 			metrics.UpdateDurationFromStart(metrics.ScaleDown, scaleDownStart)
 
 			if typedErr != nil {
 				glog.Errorf("Failed to scale down: %v", err)
 				a.lastScaleDownFailTime = currentTime
 				return typedErr
 			}
-			if result == ScaleDownNodeDeleted {
+			if scaleDownStatus.Result == status.ScaleDownNodeDeleted {
 				a.lastScaleDownDeleteTime = currentTime
 			}
+			if scaleDownStatus.Result == status.ScaleDownNodeDeleted || scaleDownStatus.Result == status.ScaleDownNodeDeleteStarted {
+				a.clusterStateRegistry.Recalculate()
```

Comment:
Do we also need to recalculate when we just started draining the node? At this point, we'll be evicting pods, but target size shouldn't yet be affected (unless the drain already completed, which is unlikely)

Reply:
That's true, I'll change it

Reply:
Done